src/main/java/unicash/model/transaction/Amount.java

```diff
@@ -26,7 +26,7 @@ public Amount(double amount) {
      * Returns true if a given amount is a non-negative value.
      */
     public static boolean isValidAmount(double amount) {
-        return amount >= 0.00;
+        return amount >= 0.00 && amount <= Integer.MAX_VALUE;
```

Comment:
I think there might be a merge conflict here, I've added the validation methods you requested together with #96